### PR TITLE
Support Async connection

### PR DESCRIPTION
**Pull request by La0**
This is needed for mozilla-releng/services support, as some part need asynchronous connection support.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 root = os.path.abspath(os.path.dirname(__file__))
 path = lambda *p: os.path.join(root, *p)
-assert os.path.exists('README.txt'), 'Missing README.txt'
-long_desc = open(path('README.txt')).read()
+assert os.path.exists('README.md'), 'Missing README.md'
+long_desc = open(path('README.md')).read()
```
Comment (klahnakoski) [inline setup.py:9]: Hmm.  The packaging process for this project involves generating a `README.txt` in restructured format from the README.md.  In theory, Pypi now supports markdown so this conversion is not needed, but it has not been tested yet.

File: esFrontLine/connection.py [hunk 2]
```diff
@@ -56,4 +57,33 @@ def perform_request(self, method, url, params, body, headers=None, *args, **kwar
...
+    def perform_request(self, method, url, params, body, headers=None, *args, **kwargs):
+        '''
+        Build a new HAWK header on each request
+        Using elasticsearch Synchronous client
+        '''
+        hawk_headers = self.add_hawk_authentication(method, url, params, body, headers)
+        return super(Urllib3HttpConnection, self).perform_request(method, url, params, body, headers=hawk_headers, *args, **kwargs)
+
+
+if ASYNC_SUPPORT:
```
Comment (klahnakoski) [inline esFrontLine/connection.py]: Personally, I would find it cleaner if the import and class definition were bundled together so `ASYNC_SUPPORT` variable is not needed.  Maybe you plan for this variable to be used in more places? 

```python
try:
    from elasticsearch_async.connection import AIOHttpConnection
    class AsyncHawkConnection(BaseHawkConnection, AIOHttpConnection):
```

No need to change anything.

**Comment by klahnakoski** [comment]
This code is not used on the server side, so I assumed this code works.  It looks fine.  Please merge it yourself.

File: esFrontLine/connection.py (hunk 2 above)
Comment (klahnakoski) [reply esFrontLine/connection.py]: Better yet, put it in its own python file, and import it only if required.

Comment (klahnakoski) [reply esFrontLine/connection.py]: For the sake of compatibility with python2, which is still around, put the asynch class in its own python file.

**Comment by klahnakoski** [comment]
Please add ?`aiohttp`? to the requirements.  Is there a 2/3 compatible way to declare requirements?

**Comment by La0** [comment]
@klahnakoski I moved the client implementations into `esFrontLine/client/`, and added some documentation for its usage (as `CLIENTS.md`).

I don't know how to add a specific requirement only for Python 2 or 3 (as `aiohttp` is only available for Python 3.4+)

**Comment by klahnakoski** [comment]
This looks much nicer!  Thank you for the doc too.

**Comment by klahnakoski** [comment]
You wrote a `tests/es_client.py` and `tests/debug_client.py`; they both need their imports updated. More important: I have not been able to run them because there is no  config file checked in. 

Please update their imports, and provide a working config.  Or, if you think my `tests/prod_client.py` is good enough (along with `tests/config/prod_client.json`), then remove both of them.

**Comment by klahnakoski** [comment]
If you want to execute multiple queries at once, using multisearch may be easier than `asynch`: http://nocf-www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html

**Comment by La0** [comment]
I removed `tests/es_client.py` and `tests/debug_client.py` and updated the import in `tests/prod_client.py`, as this one is sufficient.

**Pull request merged**